modules/db/src/main/scala/gem/dao/ObservationDao.scala

```diff
@@ -21,6 +21,7 @@ object ObservationDao {
                     ${o.id.index},
                     ${o.title},
                     ${o.instrument})
+       ON CONFLICT DO NOTHING
```

Comment:
Oh ok if you need to keep going then you have to roll back the transaction at that point.

```scala
blah.onUniqueViolation(HC.rollback as 0)
```

If you have done stuff prior to that point you can use a Savepoint to only roll back to the point before you attempt `blah`.

```scala
HC.setSavepoint.flatMap(s => blah.onUniqueViolation(HC.rollback(s) as 0))
```